src/Core/Layer/Collector/UsesCollector.php

```diff
@@ -11,18 +11,26 @@
 use Qossmic\Deptrac\Core\Ast\AstMap\AstMap;
 use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeReference;
 use Qossmic\Deptrac\Core\Ast\AstMap\ClassLike\ClassLikeToken;
+use Qossmic\Deptrac\Core\Ast\AstMapExtractor;
 
 final class UsesCollector implements CollectorInterface
 {
-    public function satisfy(array $config, TokenReferenceInterface $reference, AstMap $astMap): bool
+    private readonly AstMap $astMap;
+
+    public function __construct(private AstMapExtractor $astMapExtractor)
+    {
+        $this->astMap = $this->astMapExtractor->extract();
```

Comment:
It is to resolve this: https://github.com/qossmic/deptrac/issues/954. 
- To remove the baselined dependency of `Layer` on `Ast`.
- To remove Ast from the contract Interface, as AST itself is not part of the contract
- When passing via a constructor, you cannot pass AST directly, because it is not created at the DIC creation time. So you need to instead pass the service that can create it. I used the extractor, as it is caching the results and therefore prevent multiple builds of the `AstMap`